index.js

```diff
@@ -1,64 +1,65 @@
-const Boom     = require('boom')
-const gimme    = require('@articulate/gimme')
-const jwks     = require('jwks-rsa')
-const jwt      = require('jsonwebtoken')
+const axios                 = require('axios')
+const deepmerge             = require('deepmerge')
+const Boom                  = require('boom')
+const jwks                  = require('jwks-rsa')
+const jwt                   = require('jsonwebtoken')
+const pick                  = require('lodash.pick')
+const { promisify }         = require('util')
 const { IssWhitelistError } = require('./lib/errors')
-
-const {
-  applyTo: thrush, compose, composeP, curryN, is, isNil, ifElse,
-  merge, mergeDeepRight, partialRight, pick, prop, replace, when
-} = require('ramda')
-
-const { promisify, reject, rename, tapP } = require('@articulate/funky')
+const { rename, tapP }      = require('./lib/helpers')
 
 const wellKnown = '/.well-known/openid-configuration'
 
 const bindFunction = client =>
-  promisify(client.getSigningKey, client)
+  promisify(client.getSigningKey.bind(client))
 
 const buildClient = (jwksOpts, url) =>
-  gimme({ url })
-    .then(prop('body'))
+  axios.get(url)
+    .then(res => res.data)
     .then(rename('jwks_uri', 'jwksUri'))
-    .then(compose(jwks, merge(jwksOpts)))
+    .then(obj => jwks(Object.assign({}, jwksOpts, obj)))
     .then(bindFunction)
 
 const chooseKey = key =>
   key.publicKey || key.rsaPublicKey
 
-const decode = partialRight(jwt.decode, [{ complete: true }])
+const decode = token =>
+  jwt.decode(token, { complete: true })
 
 const enforce = token =>
   token || unauthorized('null token not allowed')
 
 const forbidden = err =>
-  reject(Boom.forbidden(err))
+  Promise.reject(Boom.forbidden(err))
 
-const stripBearer =
-  replace(/^Bearer /i, '')
+const stripBearer = str =>
+  str.replace(/^Bearer /i, '')
 
-const throwIfNull =
-  when(isNil, () => reject('invalid token'))
+const throwIfNull = val =>
+  val == null ? Promise.reject('invalid token') : val
```

Comment:
```suggestion
  val === null ? Promise.reject('invalid token') : val
```

Reply:
Soft == for "nil"

Reply:
yea js is crazy, i suggested to favor unambiguous strict equality vs relying on the idiosyncrasies in js, all good though nbd.